### PR TITLE
chain/ethereum: Remove call.input.len() validation

### PR DESCRIPTION
**Pull request by evaporei**
We don't have to check whether the call.input.len() is multiple of 32 because we now skip invalid decoding of calls.

- https://github.com/graphprotocol/graph-node/pull/3326
- ae6951328df4270e3153897220fdef1822055859

This is essentially reverting https://github.com/graphprotocol/graph-node/pull/3219 because of the new change introduced in https://github.com/graphprotocol/graph-node/pull/3326.

Related `ethabi` patch: https://github.com/graphprotocol/ethabi/commit/441c09f1e688f8c63c80dbe7ecc36fd64c745a9e

### CODE REVIEW

File: chain/ethereum/src/adapter.rs [hunk 1]
```diff
@@ -445,22 +445,11 @@ impl EthereumCallFilter {
...
-        // The first one is acting as a proxy to the second one (an atomicMatch_).
-        // If you try to decode the first call as it is/comes from a `traces` RPC request, it
-        // will fail because it has a smaller size/length.
-        let correct_input_size = (call.input.0.len() - 4) % 32 == 0;
-        if !correct_input_size {
-            return false;
-        }
-
+        // - https://github.com/graphprotocol/graph-node/pull/3326
+        // - ae6951328df4270e3153897220fdef1822055859
```
Comment (leoyvens) [inline chain/ethereum/src/adapter.rs:452]: This comment is hard to understand if you don't have all the context. Maybe this should say that decoding failures are filtered elsewhere, and have a 'see also' to that code.

Comment (evaporei) [reply chain/ethereum/src/adapter.rs:452]: Ok, I'll improve on it!

Comment (evaporei) [reply chain/ethereum/src/adapter.rs]: Do you think it's better now or do you have some other suggestion? 📝

**Comment by evaporei** [comment]
The CI is failing, I'll fix it.

**Review by leoyvens — APPROVED** [review]

File: chain/ethereum/src/data_source.rs [hunk 2]
```diff
@@ -628,6 +628,8 @@ impl DataSource {
                     },
                 ) {
                     Ok(val) => val,
+                    // Link to where the previous validation was
```
[the remark this answers is not in the document]
Comment (evaporei) [reply chain/ethereum/src/data_source.rs]: Done.

**Review by leoyvens — APPROVED** [review]
Good to squash and merge

**Comment by alcercu** [comment]
Thanks for the quick answer @leoyvens! I was asking because I thought this patch was aimed at solving this:
https://github.com/evaporei/poh-investigation
But in the hosted service version of the subgraph, the error is still present:
https://thegraph.com/hosted-service/subgraph/kleros/proof-of-humanity-mainnet

Let me know if you need more info! Thanks a lot for looking into this!

**Comment by leoyvens** [comment]
@alcercu The subgraph will unfortunately need to be resynced to be able to catch the missing calls.

**Comment by alcercu** [comment]
Perfect, thanks @leoyvens and @evaporei! 🙌🏻

**Pull request merged**